packages/stark-ui/src/modules/language-selector/components/language-selector.component.ts

```diff
@@ -0,0 +1,126 @@
+import { DateAdapter } from "@angular/material/core";
+import { Subscription } from "rxjs/Rx";
+
+import {
+	STARK_APP_METADATA,
+	StarkApplicationMetadata,
+	STARK_LOGGING_SERVICE,
+	StarkLoggingService,
+	STARK_SESSION_SERVICE,
+	StarkSessionService,
+	StarkLanguage
+} from "@nationalbankbelgium/stark-core";
+import { AbstractStarkUiComponent } from "../../../common/classes/abstract-component";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-language-selector";
+
+export type StarkLanguageSelectorMode = "dropdown" | "toolbar";
+
+/**
+ * Component to select the application's language from a list of available languages passed as parameter.
+ */
+@Component({
+	selector: "stark-language-selector",
+	templateUrl: "./language-selector.component.html",
+	encapsulation: ViewEncapsulation.None,
+	host: {
+		class: componentName
+	}
+})
+export class StarkLanguageSelectorComponent extends AbstractStarkUiComponent implements OnInit, OnDestroy {
+	/**
+	 * HTML id of action bar component.
+	 */
+	@Input()
+	public languageSelectorId: string = "";
+
+	/**
+	 * In which 'mode' the language selector is displayed: dropdown / toolbar
+	 */
+	@Input()
+	public mode: StarkLanguageSelectorMode;
+
+	/**
+	 * The currently selected language
+	 */
+	@Input()
+	public selectedLanguage: string;
+
+	/**
+	 * A reference to the sessionService subscription, needed to unsubscribe upon destroy.
+	 */
+	private sessionServiceSubscription: Subscription;
+
+	/**
+	 * Class constructor
+	 * @param appMetadata - The Metadata of the application the contains the supportedLanguages
+	 * @param logger - The logger of the application
+	 * @param sessionService - The session service of the application
+	 * @param dateAdapter - Needed to change the locale of the DateAdapter
+	 */
+	public constructor(
+		@Inject(STARK_APP_METADATA) public appMetadata: StarkApplicationMetadata,
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		@Inject(STARK_SESSION_SERVICE) public sessionService: StarkSessionService,
+		public dateAdapter: DateAdapter<any>,
+		protected renderer: Renderer2,
+		protected elementRef: ElementRef
+	) {
+		super(renderer, elementRef);
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnInit(): void {
+		this.logger.debug(componentName + ": controller initialized");
+
+		this.sessionServiceSubscription = this.sessionService
+			.getCurrentLanguage()
+			.subscribe(
+				(language: string) => (this.selectedLanguage = language),
+				() => this.logger.error(componentName + ": an error occurred getting the current language.")
+			);
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnDestroy(): void {
+		if (this.sessionServiceSubscription) {
+			this.sessionServiceSubscription.unsubscribe();
+		}
+	}
+
+	/**
+	 * Change the current language based on the selection made by the user
+	 */
+	public changeLanguage(language: string): void {
```

Comment:
As we discussed, there is a bug in the language selector that causes the language not being correctly selected in after changing the language 3 times 😕

![language selector bug](https://user-images.githubusercontent.com/5698318/45616113-cd664280-ba6e-11e8-89a8-0cfb7759dee8.PNG)